jina/serve/runtimes/gateway/grpc/gateway.py

```diff
@@ -107,6 +104,7 @@ async def shutdown(self):
 
     async def run_server(self):
         """Run GRPC server forever"""
+        await self.warmup()
```

Comment:
Can u have a task that wraps the 2 of them and await both at the same time? I am scared about the capacity to shutdown the gateway while warmup

Reply:
This is indeed an issue since the max timeout is 5 mins so until then the co-routine keeps running without cancellation. I wanted to check if there is another entry point to trigger the warmup. Currently the warmup is just triggering requests while the executors are still starting. This adds some unnecessary debug logs if there are slow executors.

Reply:
maybe u can `create` a task without awaiting it

Reply:
so that is scheduled in the event loop and in the `teardown` u make sure to cancel the task or whatever